x-pack/filebeat/input/etw/input.go

```diff
@@ -0,0 +1,215 @@
+	"github.com/elastic/beats/v7/libbeat/feature"
+	"github.com/elastic/beats/v7/x-pack/libbeat/reader/etw"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+const (
+	inputName = "etw"
+)
+
+// etwInput struct holds the configuration and state for the ETW input
+type etwInput struct {
+	log        *logp.Logger
+	config     config
+	etwSession etw.Session
+}
+
+func Plugin() input.Plugin {
+	return input.Plugin{
+		Name:      inputName,
+		Stability: feature.Beta,
+		Info:      "Collect ETW logs.",
+		Manager:   stateless.NewInputManager(configure),
+	}
+}
+
+func configure(cfg *conf.C) (stateless.Input, error) {
+	conf := defaultConfig()
+	if err := cfg.Unpack(&conf); err != nil {
+		return nil, err
+	}
+
+	return &etwInput{
+		config: conf,
+	}, nil
+}
+
+func (e *etwInput) Name() string { return inputName }
+
+func (e *etwInput) Test(_ input.TestContext) error {
+	// ToDo
+	return nil
+}
+
+// Run starts the ETW session and processes incoming events.
+func (e *etwInput) Run(ctx input.Context, publisher stateless.Publisher) error {
+	var err error
+	// Initialize a new ETW session with the provided configuration.
+	e.etwSession, err = etw.NewSession(convertConfig(e.config))
+	if err != nil {
+		return fmt.Errorf("error when initializing '%s' session: %w", e.etwSession.Name, err)
+	}
+
+	// Set up logger with session information.
+	e.log = ctx.Logger.With("session", e.etwSession.Name)
+	e.log.Info("Starting " + inputName + " input")
+
+	var wg sync.WaitGroup
+	var once sync.Once
+
+	// Handle realtime session creation or attachment.
+	if e.etwSession.Realtime {
+		if !e.etwSession.NewSession {
+			// Attach to an existing session.
+			err = e.etwSession.GetHandler()
+			if err != nil {
+				return fmt.Errorf("unable to retrieve handler: %w", err)
+			}
+			e.log.Debug("attached to existing session")
+		} else {
+			// Create a new realtime session.
+			err = e.etwSession.CreateRealtimeSession()
+			if err != nil {
+				return fmt.Errorf("realtime session could not be created: %w", err)
+			}
+			e.log.Debug("created session")
+		}
+	}
+	// Defer the cleanup and closing of resources.
+	defer func() {
+		wg.Wait() // Ensure all goroutines have finished before closing.
+		once.Do(e.Close)
+		e.log.Info(inputName + " input stopped")
+	}()
+
+	// eventReceivedCallback processes each ETW event.
+	eventReceivedCallback := func(er *etw.EventRecord) uintptr {
+		if er == nil {
+			e.log.Error("received null event record")
+			return 1
+		}
+
+		e.log.Debugf("received event %d with length %d", er.EventHeader.EventDescriptor.Id, er.UserDataLength)
+
+		var event map[string]interface{}
+
+		if data, err := etw.GetEventProperties(er); err == nil {
+			event = map[string]interface{}{
+				"Header":          er.EventHeader,
+				"EventProperties": data,
+				"Metadata":        fillEventMetadata(er, e.etwSession, e.config),
```

Comment:
I'm concerned that we are missing mappings for these fields and that they don't conform to existing conventions like being lower snake case. 

The fields produced by this need to be carefully considered on whether yield value to user (it's easier to add more fields later than it is to take away), if there is an existing ECS field that represents the data, and if there is prior art with either the winlog input or Elastic Endpoint (which sends some ETW events).